### DISCUSSION TITLE
Existing Protractor test scripts migration to

### DISCUSSION THREAD

**Discussion by rgprakash** [cypress-io/cypress#19971] [Questions and Help]
Hi all,

Does cypress has any utility to migrate existing Protractor test scripts (or) we have to manually
rewrite the protractor scripts to cypress?

Thanks!

**Comment by portyu9** [comment]
Hi @rgprakash 

There isn't an official utility that automatically converts Protractor tests to Cypress.

The tests should be migrated/re-written because Protractor and Cypress use different execution models and APIs.

For Angular projects, Cypress does provide a schematic to install and configure Cypress:

```bash
ng add @cypress/schematic
```

Then migrate the tests incrementally, for example:

```js
// Protractor
element(by.css('#email')).sendKeys('test@example.com')

// Cypress
cy.get('#email').type('test@example.com')
```

You do not need to migrate everything at once—Protractor and Cypress can coexist while you convert the suite gradually.

**Reply by portyu9** [reply]
@rgprakash Please mark this answer as accepted if it helped, thank you.
